### PR TITLE
feat: add onChange override function

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*

*Description of changes:*
- add `onChange` on form types
- call `onChange` if the prop is passed down
- slight refactor on how variable statements are created

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts [hunk 1]
```diff
@@ -480,12 +506,67 @@ export function buildOnBlurStatement(fieldName: string) {
...
+            ],
+            NodeFlags.Const,
+          ),
+        ),
+        factory.createExpressionStatement(
+          factory.createBinaryExpression(
+            factory.createIdentifier('value'),
+            factory.createToken(SyntaxKind.EqualsToken),
+            factory.createBinaryExpression(
+              buildAccessChain(['result', ...fieldName.split('.')]),
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:551]: this is to build the access chain to support nested json (example in snapshot diff and below)
```js
value = result?.bio?.favoriteAnimal ?? value;
```

File: packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap [hunk 2]
```diff
@@ -6,6 +6,7 @@ exports[`form-render utils should generate before & complete types if datastore
     onSuccess?: (fields: Record<string, unknown>) => void;
     onError?: (fields: Record<string, unknown>, errorMessage: string) => void;
     onCancel?: () => void;
+    onChange?: (fields: Record<string, unknown>) => Record<string, unknown>;
```
Comment (cwoolum) [inline packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap:9]: Shouldn't the type be specifically related to the fields in the form? We should be able to infer all of those.

**Review by cwoolum — COMMENTED** [review]
One question

File: packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap (hunk 2 above)
Comment (SwaySway) [reply packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap:9]: I can create a follow up pr to create the types for datastore-backed forms, though for custom forms it's bit more unknown as we'd need a larger task to track those data types.
cc: @hein-j

Comment (cwoolum) [reply packages/codegen-ui-react/lib/__tests__/react-forms/__snapshots__/form-renderer-helper.test.ts.snap:9]: Even for JSON forms though, we still know the types that the different primitives return so we can still provide a map of fields to those types.

**Review by cwoolum — APPROVED** [review]
Please follow up with the proper types in a later CR

**Review by hein-j — APPROVED** [review]

**Pull request merged**